File: test/new_lang/test4/test_program.cpp
#include <splashkit.h>

#include <iostream>
using namespace std;

int main()
{
  int i;
  uint ui;
  float f;
  double d;
  short sh;
  unsigned short ush;
  int64_t l;
  char ch;
  unsigned char uch;
  bool b;

  i = 10;
  f = 1.11f;
  d = 2.22;
  sh = -3;
  ush = 4;
  l = 5;
  ch = 'a';
  uch = 'B';
  b = true;
  ui = 50;

  cout << i << " get " << get_and_update_int(i) << " updated int " << i << endl;

  cout << i << " ";
  update_int(i);
  cout << "Updated int: " << i << endl;

  cout << ui << " ";
  update_uint(ui);
  cout << "Updated uint: " << ui << endl;

  cout << sh << " ";
  update_short(sh);
  cout << "Updated short int: " << sh << endl;

  cout << ush << " ";
  update_ushort(ush);
  cout << "Updated short int: " << ush << endl;

  cout << f << " ";
  update_float(f);
  cout << "Updated float: " << f << endl;

  cout << d << " ";
  update_double(d);
  cout << "Updated double: " << d << endl;

  cout << l << " ";
  update_long(l);
  cout << "Updated long: " << l << endl;

  cout << ch << " ";
  update_char(ch);
  cout << "Updated char: " << ch << endl;

  cout << uch << " ";
  update_uchar(uch);
  cout << "Updated char: " << uch << endl;

  cout << b << " ";
  update_bool(b);
  cout << "Updated bool: " << b << endl;
}

File: test/new_lang/test4/test_program.py
from splashkit import *

i = c_int(10)
f = c_float(1.11)
d = c_double(2.22)
sh = c_short(-3)
ush = c_ushort(4)
l = c_long(5)
ch = c_char(b'a')
uch = c_ubyte(ord('B'))
b = c_bool(True)
ui = c_uint(50)

print(i)
print(" get ", get_and_update_int(i), " updated int ", i)

print(i, end=" ")
update_int(i)
print("Updated int: ", i)

print(ui, end=" ")
update_uint(ui)
print("Updated uint: ", ui)

print(sh, end=" ")
update_short(sh)
print("Updated short int: ", sh)

print(ush, end=" ")
update_ushort(ush)
print("Updated short int: ", ush)

print(f, end=" ")
update_float(f)
print("Updated float: ", f)

print(d, end=" ")
update_double(d)
print("Updated double: ", d)

print(l, end=" ")
update_long(l)
print("Updated long: ", l)

print(ch, end=" ")
update_char(ch)
print("Updated char: ", ch)

print(uch, end=" ")
update_uchar(uch)
print("Updated char: ", uch)

print(b, end=" ")
update_bool(b)
print("Updated bool: ", b)
